src/filemanager.cpp

```diff
@@ -0,0 +1,2301 @@
+                         pj_curl_write_func);
+
+        body.clear();
+        body.reserve(size_to_read);
+        curl_easy_setopt(hCurlHandle, CURLOPT_WRITEDATA, &body);
+        curl_easy_setopt(hCurlHandle, CURLOPT_WRITEFUNCTION,
+                         pj_curl_write_func);
+
+        file->m_szCurlErrBuf[0] = '\0';
+
+        curl_easy_perform(hCurlHandle);
+
+        long response_code = 0;
+        curl_easy_getinfo(hCurlHandle, CURLINFO_HTTP_CODE, &response_code);
+
+        curl_easy_setopt(hCurlHandle, CURLOPT_HEADERDATA, nullptr);
+        curl_easy_setopt(hCurlHandle, CURLOPT_HEADERFUNCTION, nullptr);
+
+        curl_easy_setopt(hCurlHandle, CURLOPT_WRITEDATA, nullptr);
+        curl_easy_setopt(hCurlHandle, CURLOPT_WRITEFUNCTION, nullptr);
+
+        if (response_code == 0 || response_code >= 300) {
+            const double delay =
+                GetNewRetryDelay(static_cast<int>(response_code), oldDelay,
+                                 body.c_str(), file->m_szCurlErrBuf);
+            if (delay != 0 && delay < MAX_RETRY_DELAY_MS) {
+                pj_log(ctx, PJ_LOG_TRACE,
+                       "Got a HTTP %ld error. Retrying in %d ms", response_code,
+                       static_cast<int>(delay));
+                proj_sleep_ms(static_cast<int>(delay));
+                oldDelay = delay;
+            } else {
+                if (out_error_string) {
+                    if (file->m_szCurlErrBuf[0]) {
+                        snprintf(out_error_string, error_string_max_size, "%s",
+                                 file->m_szCurlErrBuf);
+                    } else {
+                        snprintf(out_error_string, error_string_max_size,
+                                 "HTTP error %ld: %s", response_code,
+                                 body.c_str());
+                    }
+                }
+                return nullptr;
+            }
+        } else {
+            break;
+        }
+    }
+
+    if (out_error_string && error_string_max_size) {
+        out_error_string[0] = '\0';
+    }
+
+    if (!body.empty()) {
+        memcpy(buffer, body.data(), std::min(size_to_read, body.size()));
+    }
+    *out_size_read = std::min(size_to_read, body.size());
+
+    file->m_headers = std::move(headers);
+    return reinterpret_cast<PROJ_NETWORK_HANDLE *>(file.release());
+}
+
+// ---------------------------------------------------------------------------
+
+static void pj_curl_close(PJ_CONTEXT *, PROJ_NETWORK_HANDLE *handle,
+                          void * /*user_data*/) {
+    delete reinterpret_cast<CurlFileHandle *>(handle);
+}
+
+// ---------------------------------------------------------------------------
+
+static size_t pj_curl_read_range(PJ_CONTEXT *ctx,
+                                 PROJ_NETWORK_HANDLE *raw_handle,
+                                 unsigned long long offset, size_t size_to_read,
+                                 void *buffer, size_t error_string_max_size,
+       
```

Comment:
In my opinion, filemanager.cpp is too long and has too many classes in it. You're going to have to search hard to find specific things unless you know what you're looking for. I prefer things broken up a little bit more, but that's a preference thing.

Reply:
Yeah, it is a big one. Perhaps the bits that has to do with networking can be moved to a separate file? `networkfilemanager.cpp` perhaps?

Reply:
All network related stuff moved to networkfilemanager.cpp